lib/memory.cpp

```diff
@@ -75,6 +77,7 @@ void memory::pmemory_t::write_byte(uint64_t addr, uint8_t data) {
             break;
         default:
             // Not MMIO
+            assert((addr < RAMSIZE) && "Attempt to write out of bounds; did you call check_writable_byte() first?");
```

Comment:
I was thinking instead of duplicating the checks it would be easier/less mistake prone if this function just called check_writable_byte() first and if it threw an exception then it would assert false

Reply:
Ah yes my bad, I will remove that check (that's a much better idea and I didn't realize you had already done so in your branch)

Reply:
Alrighty I've removed that line